lib/serializer.ts

```diff
@@ -669,9 +669,10 @@ function getPolymorphicMapper(serializer: Serializer, mapper: CompositeMapper, o
     if (discriminatorName != undefined) {
       const discriminatorValue = object[discriminatorName];
       if (discriminatorValue != undefined) {
-        const indexDiscriminator = discriminatorValue === mapper.type.uberParent
+        const parent = mapper.type.uberParent || mapper.type.className;
```

Comment:
Please rename `parent` to `typeName`. `uberParent` and `className` may actually point to the current type.